src/arch/linux/mapping/mapping.c

```diff
@@ -906,16 +906,18 @@ static dosaddr_t do_get_hardware_ram(unsigned addr, uint32_t size,
 	struct hardware_ram **r_hw)
 {
   struct hardware_ram *hw;
+  dosaddr_t va = (dosaddr_t)-1;
 
   for (hw = hardware_ram; hw != NULL; hw = hw->next) {
-    if (hw->vbase != -1 &&
-	hw->base <= addr && addr + size <= hw->base + hw->size) {
-	if (r_hw)
-	  *r_hw = hw;
-      return hw->vbase + addr - hw->base;
+    if (hw->base <= addr && addr + size <= hw->base + hw->size) {
+      if (hw->vbase != (dosaddr_t)-1)
```

Comment:
So you can bypass initializing va and
return -1, and yet have some side-effect
via passed pointer r_hw?
That's a huge programming anti-pattern.

Reply:
I've now switched hw and va here, to eliminate that priority issue, and as a bonus it can then be easily used in `get_hardware_uaddr()` as well.